app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -83,6 +83,9 @@ def edit
         if page_is_locked?
           flash[:notice] = Alchemy.t('This page is locked', name: @page.locker_name)
           redirect_to admin_pages_path
+        elsif user_has_invalid_scope
+          flash[:notice] = Alchemy.t('Incorrect user type for this page')
```

Comment:
The flash message should be a bit more explaining. Maybe something like:

"Your user role does not allow you to edit this page."